### PR TITLE
Complete x-nucleo-iks01a2 shield support

### PR DESCRIPTION
**Pull request by avisconti**
The x-nucleo-iks01a2 shield is a STM sensor board with Arduino connector.
This PR includes following enhancement:

1. Add LSM303AGR support in x-nucleo-iks01a2 shield
2. Add x-nucleo-iks01a2 shield sample code for testing

### CODE REVIEW

File: samples/shields/x_nucleo_iks01a2/README.rst
Comment (dbkinder) [inline samples/shields/x_nucleo_iks01a2/README.rst]: Could you provide a hint (or reference material) that describes how to configure the I2C this way (references to files in this PR or links to external documentation)?

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_f401re/nucleo_f401re.dts
Comment (erwango) [inline boards/arm/nucleo_f401re/nucleo_f401re.dts]: This requires doc update, but same change done in #12334. You can drop this commit

File: samples/shields/x_nucleo_iks01a2/src/main.c
Comment (erwango) [inline samples/shields/x_nucleo_iks01a2/src/main.c]: Use ST copyright

**Review by erwango — CHANGES_REQUESTED** [review]
2 minor comments

File: boards/arm/nucleo_f401re/nucleo_f401re.dts
Comment (avisconti) [reply boards/arm/nucleo_f401re/nucleo_f401re.dts]: ack

File: samples/shields/x_nucleo_iks01a2/src/main.c
Comment (avisconti) [reply samples/shields/x_nucleo_iks01a2/src/main.c]: ack

File: samples/shields/x_nucleo_iks01a2/README.rst
Comment (avisconti) [reply samples/shields/x_nucleo_iks01a2/README.rst]: OK, I will add reference to the Nucleo F401RE board documentation, which is the one I used
to test the shield sample. The documentation has been updated by @erwango to include the
arduino stuff description and will be merged soon.

**Comment by avisconti** [comment]
@erwango @dbkinder 
Repushed taking care of your requests.
Even if this PR does not strictly depend from #12334 I think it would be better to wait until that one
is merged, since I made explicit reference to that board both as a test veichle and as a
reference to add arduino support.

**Review by erwango — APPROVED** [review]

**Comment by avisconti** [comment]
@dbkinder 
Can you confirm documentation is now in the proper form?
It seems to me I covered all the points reported.

**Comment by avisconti** [comment]
@erwango 
> we're changing the way shields are handled, this will have impact on your PR, you might 
> want to have a look: #12403

Sure, I will change the PR accordingly. I'll wait for #12403 to be merged.

**Review by dbkinder — APPROVED** [review]
OK on docs, thanks.

**Review by galak — CHANGES_REQUESTED** [review]
Marking this a 'request change' to have it pend on the -DSHIELD rework PR.

**Comment by avisconti** [comment]
> Marking this a 'request change' to have it pend on the -DSHIELD rework PR.

Sure, I'll wait til that work is merged.

**Comment by avisconti** [comment]
re-pushed keeping in mind the -DSHIELD stuff (#12403).
Please note that I also made this PR dependent from  #12623 and #12373.

**Comment by erwango** [comment]
@avisconti, #12403 has been merged, you can rebase on master to use it

**Comment by avisconti** [comment]
> @avisconti, #12403 has been merged, you can rebase on master to use it

Yes, already did.
Just waiting for  #12623 and #12373.

**Review by ydamigos — APPROVED** [review]

**Comment by avisconti** [comment]
Rebased and repushed after #12623 and #12373 merging.

**Comment by avisconti** [comment]
@galak 
Can you looka t it once more?
It is in line with the work done for x_nucleo_ikso1a1 (-DSHIELD).

**Review by galak — APPROVED** [review]

**Pull request merged**
